packages/davinci-client/src/lib/davinci.types.ts

```diff
@@ -46,31 +46,35 @@ export interface Links {
   };
 }
 
-export type Combobox = {
-  inputType: 'MULTI_SELECT';
+export type StandardFieldValue = {
+  type:
+    | 'PASSWORD'
+    | 'TEXT'
+    | 'SUBMIT_BUTTON'
+    | 'FLOW_BUTTON'
+    | 'FLOW_LINK'
+    | 'SOCIAL_LOGIN_BUTTON'
+    | 'BUTTON';
   key: string;
   label: string;
-  options: {
-    label: string;
-    value: string | number;
-  }[];
-  required?: boolean;
-  type: 'COMBOBOX' | 'DROPDOWN';
+
+  // Optional properties
+  links?: Links;
 };
 
-export type Radio = {
-  inputType: 'SingleSelect';
+export type SingleSelect = {
```

Comment:
gotcha so these are "the same" type.

I wonder how this narrows, i'll read through a test.

I think it should narrow without a generic but my brain is questioning it